cpp/src/parquet/column_reader.cc

```diff
@@ -987,6 +993,7 @@ class TypedRecordReader : public ColumnReaderImplBase<DType>,
   std::shared_ptr<ResizableBuffer> ReleaseValues() override {
     if (uses_values_) {
       auto result = values_;
+      PARQUET_THROW_NOT_OK(result->Resize(bytes_for_values(values_written_), true));
```

Comment:
Use `/*shrink_to_fit=*/false` here? Downstream application code should not read beyond `result->size()` (e.g. any IPC code will not) and avoiding the realloc may yield better performance in some cases

Reply:
I've checked and we're using `shrink_to_fit = true` in Arrow builders. I'm not sure why we would do things differently here, especially as it may waste significant amounts of space.

Reply:
Alright, that's fine then.